tools/SourceKit/tools/sourcekitd/lib/API/CodeCompletionResultsArray.cpp

```diff
@@ -105,6 +109,7 @@ class CodeCompletionResultsArray {
     const char *DocBrief;
     const char *AssocUSRs;
     sourcekitd_uid_t SemanticContext;
+    sourcekitd_uid_t TypeRelation;
```

Comment:
Would it be worth compressing this to an integral value? @akyrtzi  suggested that we split out `unknown` (probably treated as `nil` at the protocol layer) from `unrelated`, which would give us 5 total cases or 3 bits.  For a further improvement, semantic context only uses 3 bits as well and we could pack them both into one byte.  This format is entirely internal, so we can change it if we need more cases in the future.  WDYT?

Reply:
Good idea. Since this is an optimization purely inside SourceKit, I will do it in follow-ups